### ISSUE TITLE
Issue with rearrangeTopFolders

### ISSUE THREAD

**Issue by nmargono** [whitesource/unified-agent-distribution#60]
I have a Maven-based repo, with a root pom.xml. Maven version used is 3.6.0.

Was using WS Agent `24.8.1` before and it worked fine scanning my repo.

Currently trying to upgrade to use WS Agent 24.10.1 and it is failing during Maven's PreStep and when I set `log.level=debug`, I see that it's because the agent changed the topFolder: `[DEBUG] [2024-11-15 21:47:32,813 +0000] - rearrangeTopFolders - trying to identify the root poms and their submodules` 

This, in turn, will select the wrong topFolder and spew a red herring error.

I have tried to set `rearrangeTopFolders=false`, but the Agent still insists on changing the topFolder.

Is there any way to circumvent this?

**Comment by jssap** [comment]
This also occurs in version `24.10.2`.

**Comment by nmargono** [comment]
> This also occurs in version `24.10.2`.

Yes, but as of yesterday, this was still in "open beta", so I thought I'd give it some grace 😃

**Comment by jssap** [comment]
What's the latest stable version?

**Comment by nmargono** [comment]
For us, `24.8.1` is the latest stable that passed; anything after that gives the same error.

I did notice they released `24.10.3` yesterday, but I still can't download it https://unified-agent.s3.amazonaws.com/wss-unified-agent-24.10.3.jar -- when it's up I will give it a try.

**Comment by nmargono** [comment]
Confirmed that it's still _not_ working with the latest 24.10.3:
```
[INFO] [2024-11-19 19:00:13,627 +0000] - UnifiedAgent version (pluginVersion) : 24.10.3
[...]
[DEBUG] [2024-11-19 19:00:17,479 +0000] - rearrangeTopFolders - trying to identify the root poms and their submodules
```

Hoping this can get addressed soon...we want to be on the latest for security, but this is a blocker.

**Comment by jssap** [comment]
Confirming, that `24.8.1` is the latest version that does not fail. Will have to use this until it's resolved.

**Comment by NitzanOrK** [comment]
Thank you for bringing this to our attention. To investigate and address the issue effectively, we kindly ask you to open a support ticket and include the relevant logs. This will help us resolve the matter promptly and efficiently.

**Comment by nmargono** [comment]
@NitzanOrK apologies, it's my first time here, how do I go about creating a support ticket? Can you share a link? I tried to find one through the official website, but this GH repo is the closest thing I could find in getting help.

**Comment by NitzanOrK** [comment]
@nmargono Sure! This is the link to our support portal: https://whitesourcesoftware.my.site.com/SupportPortal/s/login/

**Comment by NitzanOrK** [comment]
@nmargono Please let me know if you need assistance creating a support ticket.

**Comment by jssap** [comment]
@NitzanOrK  will this issue be closed once a version fixing this has been released?

**Comment by NitzanOrK** [comment]
@jssap First, we’d like to verify whether this is a technical issue or related to another matter. If it’s a technical problem, we’ll address it and communicate the issue in our release notes. If it’s something else, our support team will be able to assist further.

To ensure proper investigation, we kindly ask that a support ticket be created. I’d be happy to assist you in creating the ticket if needed—just let me know(:

**Comment by nmargono** [comment]
@NitzanOrK I'm trying to get in, but it looks like I need an approver to create a Support user on my company side...or is there another way to create a Support ticket without having to do that?

**Comment by NitzanOrK** [comment]
@nmargono Sure, you can also use this email to open a ticket: support@mend.io. Our support team will take if from there. 
If you have any additional requests or questions I'd be happy to assist.

**Comment by nmargono** [comment]
@NitzanOrK I have created a Mend Support Case 00145151 -- hope someone picks it up soon!

**Comment by NitzanOrK** [comment]
@nmargono Great. I'll check with support. If you don't get a response within a few days, you can contact me here.
